examples/custom_search_method/core_search_runner/run_experiment.py

```diff
@@ -0,0 +1,67 @@
+"""
+This script runs a custom SearchMethod with CoreSearchRunner on a Determined cluster.
+
+CoreSearchRunner is responsible for:
+ -> executing the custom SearchMethod as a single trial experiment on the Determined cluster,
+ -> creating a multi-trial experiment on the Determined cluster,
+ -> handling communication between the multi-trial experiment and the custom SearchMethod,
+ -> enabling fault tolerance for SearchMethods that implement save_method_state() and
+    load_method_state().
+
+CoreSearchRunner receives SearcherEvents from the multi-trial experiment, and passes
+the events to your custom SearchMethod, which, in turn, produces a list of Operations.
+Next, CoreSearchRunner sends the operations to the multi-trial experiment for execution.
+"""
+
+import logging
+import determined as det
+from asha import ASHASearchMethod
+from determined.searcher.core_search_runner import CoreSearchRunner
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO, format=det.LOG_FORMAT)
+
+
+    ########################################################################
+    # Multi-trial experiment
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    #
+    # The content of the following directory is uploaded to Determined cluster.
+    # It should include all files necessary to run the experiment (as usual).
+    model_context_dir = 'experiment_files'
+
+    # Path to the .yaml file with the multi-trial experiment configuration.
+    model_config = "experiment_files/custom_config.yaml"
+
+    ########################################################################
+    # Fault Tolerance for CoreSearchRunner
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    #
+    # To support fault tolerance, CoreSearchRunner saves information required to
+    # resume CoreSearchRunner and SearchMethod in the case of unexpected process termination.
+    # The required information is stored in CoreSearchRunner experiment checkpoint (reported through Core API
+    # checkpointing) and includes experiment id, CoreSearchRunner state, and SearchMethod state.
+    # To learn more about checkpointing, see https://docs.determined.ai/latest/reference/python-api.html#checkpoint
+    #
+    # While CoreSearchRunner saves its own state and ensures invoking save() and
+    # load() methods when necessary, a user is responsible for implementing
+    # SearchMethod.save_method_state() and SearchMethod.load_method_state() to ensure correct
+    # resumption of the SearchMethod.
+    with det.core.init() as core_context:
+
+        # Instantiate your implementation of SearchMethod
+        search_method = ASHASearchMethod(max_length=1000, max_trials=16, num_rungs=3, divisor=4)
```

Comment:
let's put these into the `hyperparameters` section of the config below and get them from experiment `info`